packages/neuron-wallet/src/startup/loopTask/create.ts

```diff
@@ -1,78 +1,67 @@
-
-const loadURL = `file://${path.join(__dirname, 'index.html')}`
 
-const stopLoopSubject = new Subject()
+networkSwitchSubject.subscribe(async (network: NetworkWithID | undefined) => {
+  if (network) {
+    await initConnection(network.name)
+  }
+})
 
 // TODO: mock as an address subject
-const addressChangeSubject = new Subject()
-
-// maybe should call this every time when new address generated
-// load all addresses and convert to lockHashes
-const loadAddressesAndConvert = async (): Promise<string[]> => {
-  const addresses: string[] = Address.allAddresses().map(addr => addr.address)
-  const lockHashes: string[] = await Promise.all(
-    addresses.map(async addr => {
-      return LockUtils.addressToLockHash(addr)
-    }),
-  )
-  return lockHashes
-}
-
-// call this after network switched
-// TODO: listen to network switch
-const switchNetwork = async (networkId: string) => {
-  // stop all blocks service
-  stopLoopSubject.next('stop')
-  // disconnect old connection and connect to new database
-  await initConnection(networkId)
-  // load lockHashes
-  const lockHashes: string[] = await loadAddressesAndConvert()
-  // start sync blocks service
-  const syncBlocksService = new SyncBlocksService(lockHashes)
+export const addressChangeSubject = new Subject()
 
-  // TODO: should change to listen real address module event
-  addressChangeSubject.subscribe(async () => {
-    syncBlocksService.lockHashes = await loadAddressesAndConvert()
-  })
-
-  stopLoopSubject.subscribe(() => {
-    syncBlocksService.stop()
-  })
+const loadURL = `file://${path.join(__dirname, 'index.html')}`
 
-  await syncBlocksService.loopBlocks()
+// pass data in main process to renderer process
+const syncTaskAttrs = {
+  networkSwitchSubject,
+  nodeService,
+  addressChangeSubject,
 }
 
+Object.defineProperty(app, 'syncTask', {
+  get: () => {
+    return syncTaskAttrs
+  },
+})
+
+/* eslint global-require: "off" */
 // create a background task to sync transactions
 // this task is a renderer process
 const createLoopTask = () => {
+  const windowState = windowStateKeeper({
+    defaultWidth: 1366,
+    defaultHeight: 768,
+  })
   let loopWindow: BrowserWindow | null = new BrowserWindow({
-    x: 10,
-    y: 10,
+    x: windowState.x,
+    y: windowState.y,
+    width: windowState.width,
+    height: windowState.height,
+    minWidth: 800,
+    minHeight: 600,
     show: false,
+    backgroundColor: '#e9ecef',
     webPreferences: {
-      nodeIntegration: false,
+      nodeIntegration: true,
     },
   })
 
   loopWindow.loadURL(loadURL)
 
   loopWindow.on('ready-to-show', async () => {
-    loopWindow!.hide()
-
-    // TODO: add an event on networkId for when it init and changed, it should broadcast a message with networkId
-    networkSwitchSubject.subscribe(async network => {
-      if (network) {
-        await switchNetwork(network.name)
-      }
-    })
+    if (env.isDevMode && process.env.DEV_SYNC_TASK) {
+      loopWindow!.show()
```

Comment:
Is this for local debug purpose? And why do we need to persist the state of the loop window, despite the fact this will be a hidden window for background process?

BTW, `loopWindow` doesn't describe its responsibility (e.g. `txSyncBackgroundWindow`) but its implementation detail (why the outer modules need to know there would be a loop inside?).

Reply:
Yes, it's for debug.(database log will show in window console)
It's right we should remove state for window.
Maybe `loopWindow` is not a good name, I'll rename it.